x/name/keeper/msg_server.go

```diff
@@ -128,11 +128,23 @@ func (s msgServer) DeleteName(goCtx context.Context, msg *types.MsgDeleteNameReq
 		return nil, sdkerrors.ErrUnauthorized.Wrap("msg sender cannot delete name")
 	}
 	// Delete
-	if err := s.Keeper.DeleteRecord(ctx, name); err != nil {
+	err = s.Keeper.DeleteRecord(ctx, name)
+	if err != nil {
 		ctx.Logger().Error("error deleting name", "err", err)
 		return nil, sdkerrors.ErrInvalidRequest.Wrap(err.Error())
 	}
 
+	// Remove all attributes from assigned accounts
+	accts, err := s.Keeper.attrKeeper.AccountsByAttribute(ctx, name)
```

Comment:
My only recommendation is to extract this block out into a new function, but that's mainly stylistic.

RemoveAllAttributes(ctx, name)